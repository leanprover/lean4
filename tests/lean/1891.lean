namespace Ex1
  class FunLike (F : Sort _) (β : outParam <| Nat → Sort _) where
    coe : F → ∀ a, β a

  inductive Secret
  def Wrapper := Secret
  inductive Bla | z
  instance : FunLike Bla (fun _ => Wrapper) := sorry

  instance (priority := 100) {F β} [FunLike F β] :
    CoeFun F fun _ => ∀ a : Nat, β a where coe := FunLike.coe

  #check Bla.z ∘ id
end Ex1


namespace Ex2
structure Secret
def Wrapper := Secret
def f (a : Nat) : (fun _ => Wrapper) a := ⟨⟩

#check f ∘ id
end Ex2
