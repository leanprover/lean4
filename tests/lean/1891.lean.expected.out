1891.lean:8:2-8:10: warning: declaration uses 'sorry'
FunLike.coe Bla.z ∘ id : Nat → Wrapper
f ∘ id : Nat → Wrapper
